electron/menu/index.ts

```diff
@@ -0,0 +1,175 @@
+import { ConnectionStoreItem } from "@/lib/conn-manager-store";
+import { app, Menu, MenuItemConstructorOptions, shell } from "electron";
+import { isMac } from "../utils";
+import { createDatabaseWindow, windowMap } from "../window/create-database";
+import { OuterbaseApplication } from "../type";
+import { createWindow } from "../main";
+import { createConnectionWindow } from "../window/create-connection-window";
+import { OUTERBASE_GITHUB, OUTERBASE_WEBSITE } from "../constants";
+
+export function createMenu(
+  win: OuterbaseApplication["win"],
+  connections: ConnectionStoreItem[],
+) {
+  function handleClick() {
+    createWindow();
+  }
+
+  function onOpenConnectionWindow(type: ConnectionStoreItem["type"]) {
+    createConnectionWindow(win, type);
+    win?.hide();
+  }
+
+  function generateSubMenu() {
+    const connMenu: MenuItemConstructorOptions["submenu"] = connections.map(
+      (conn) => {
+        return {
+          label: conn.name,
+          click: () => {
+            const existingWindow = windowMap.get(conn.id);
+            if (existingWindow && !existingWindow.isDestroyed()) {
+              existingWindow.focus();
+            } else {
+              createDatabaseWindow({ win, conn });
+              win?.hide();
+            }
+          },
+        };
+      },
+    );
+
+    return connMenu;
+  }
```

Comment:
Make sure you slice the connection let say last 10, if there is more than 10, you can put "See more connections"